.github/workflows/build-and-test.yml

```diff
@@ -113,13 +113,16 @@ jobs:
     needs: [ swig_Linux ]
     runs-on: ubuntu-20.04
     if: ${{ github.event_name == 'pull_request' }}
+    strategy:
+      matrix:
+        python-version: [3.6.8, 3.7.9, 3.8.8, 3.9.2]
```

Comment:
We might prefer sticking to major releases: `3.6`, `3.7`, etc, as we noted that `setup-python` when requested with a precise version often had cache miss and required extra API calls, that would impair our rate limit